packages/connectivity/ios/Classes/ConnectivityPlugin.m

```diff
@@ -140,4 +184,18 @@ - (FlutterError*)onCancelWithArguments:(id)arguments {
   return nil;
 }
 
+#pragma mark - CLLocationManagerDelegate
+
+- (CLLocationManager *)locationManager {
+ if (!_locationManager) {
+   _locationManager = [CLLocationManager new];
+   _locationManager.delegate = self;
+ }
+ return _locationManager;
+}
+
+- (void)locationManager:(CLLocationManager *)manager didChangeAuthorizationStatus:(CLAuthorizationStatus)status {
+  [self requestLocationServiceAuthorizationIfUndetermined:self.result always:nil];
```

Comment:
Should probably set `self.result` to `nil` here so we don't accidentally notify it more than once.